### PR TITLE
4.4 Generate directions from Calendar

### PR DESCRIPTION
**Pull request by HoangVuLuu**
## Description:
In this PR, I added functionality to the buttons inside Calendar to get direction to the building, directions to the room and the saved button. 
## Related User Story:
[(Link the related task)
](https://github.com/Mini-Capstone-Fried-Chicken/Fried-Chicken/issues/33)
## Screenshot/Demo:
image

CLICKING ON "GO TO BUILDING" -->

image

CLICKING ON "GO TO ROOM" -->

image

## How to test:
1. Log in the app
2. Add your google Calendar using your Gmail
3. Make sure you have a calendar set up correctly (with an event that has a building and a room)
4. Once you click on that event
5. Click on go to building
6. Click on go to room
7. click on saved
## Type of Change:
- [x] Feature
- [ ] Bug fix
- [ ] Refactor
- [ ] Documentation update
- [x] Tests
- [ ] Others

## Pull Request Checklist
- [x] Flutter CI test passed
- [x] Sonar test passed
- [x] Codecov test passed
- [x] Application runs without errors
- [x] Feature tested locally on a high level
- [x] Architecture conventions respected
- [x] No API keys or secrets committed
- [x] Document when necessary

## PR Review Checklist (dev)
- [ ] Code is readable and well structured
- [ ] No obvious bugs or edge cases missed
- [ ] Tests are adequate
- [ ] Naming and organization are clear
- [ ] Feature works as described
- [ ] No unnecessary complexity
- [ ] Follows project architecture

## PR Review Checklist (QA)
- [ ] Made sure to test the application fully
- [ ] No bugs created by this PR

### CODE REVIEW

File: lib/features/saved/saved_directions_controller.dart [hunk 1]
```diff
@@ -1,16 +1,69 @@
...
-  static final ValueNotifier<SavedPlace?> notifier = ValueNotifier<SavedPlace?>(null);
+  static final ValueNotifier<SavedPlace?> notifier = ValueNotifier<SavedPlace?>(
+    null,
+  );
 
   static void requestDirections(SavedPlace place) {
     notifier.value = place;
   }
 
+  static BuildingPolygon? _findBuildingByCode(String buildingCode) {
```
Comment (maddie-luu) [inline lib/features/saved/saved_directions_controller.dart]: One suggestion for potential improvement is the repeated case conversaion inside this method. Right now code.toUpperCase() is recomputed on every loop iteration, and item.code.toUpperCase() is also recalculated each time. It would be a bit cleaner and more efficient to normalize buildingCode once before the loop, then compated against a normalized item code. If this lookup happens often, it may even be worth storing buildings in a map keyed by normalized code instead of doing a linear scan every time.

File: lib/features/saved/saved_directions_controller.dart [hunk 2]
```diff
@@ -1,16 +1,69 @@
...
+      openingHoursToday: 'Hours unavailable today',
+      roomCode: roomCode?.trim().isEmpty == true ? null : roomCode?.trim(),
+    );
+  }
+
+  /// Convenience helper to request directions to a Concordia building by its code.
+  ///
+  /// This reuses the existing directions pipeline in `OutdoorMapPage` which listens
+  /// to [notifier] and starts a route preview from the user's current location.
+  static void requestDirectionsToBuildingCode(String buildingCode) {
```
Comment (maddie-luu) [inline lib/features/saved/saved_directions_controller.dart]: One small note is that the failure path is silent. Returning early is safe, but it may make debugging harder if an invalid building code is passed in. Depending on the surrounding architecture, it could be useful to log, assert, or expose some feedback when no building is found.

File: lib/features/saved/saved_directions_controller.dart [hunk 3]
```diff
@@ -1,16 +1,69 @@
...
+    String? roomCode,
+  }) {
+    return SavedPlace(
+      id: building.code,
+      name: building.name.isNotEmpty ? building.name : building.code,
+      category: 'all',
+      latitude: building.center.latitude,
+      longitude: building.center.longitude,
+      openingHoursToday: 'Hours unavailable today',
+      roomCode: roomCode?.trim().isEmpty == true ? null : roomCode?.trim(),
```
Comment (maddie-luu) [inline lib/features/saved/saved_directions_controller.dart]: This is a little dense, it can be rendered more readable by trimming once into a local variable and then checking whether it's empty. It would also avoid calling trim() twice.

**Review by maddie-luu — APPROVED** [review]

**Review by HaichuanLi — APPROVED** [review]
